subprojects/core/src/integTest/groovy/org/gradle/api/invocation/GradleLifecycleAllprojectsIntegrationTest.groovy

```diff
@@ -0,0 +1,293 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.invocation
+
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+import org.gradle.test.precondition.Requires
+import org.gradle.test.preconditions.IntegTestPreconditions
+
+@Requires(IntegTestPreconditions.NotIsolatedProjects)
+class GradleLifecycleAllprojectsIntegrationTest extends AbstractIntegrationSpec {
+
+    def 'lifecycle.allprojects is not executed again after project evaluation'() {
```

Comment:
💭  I am not sure the test checks what it says in the name. We'd need to check the number of occurrences of the strings in the output to verify this claim. Should we adjust the test name somehow?

Reply:
> Also, I don't think that checking the number of occurrences is the only way to verify this behaviour.
Certainly, many things would do, though we currently don't do any 🙃 

> Did you mean having a fixture like outputContains(string :"foo", times : 1) ?
That's an idea, but it does not have to be that

Reply:
> though we currently don't do any

Could you explain, why current output verification doesn't conform test scenario name?

Reply:
Let me clarify that I was nitpicking. Nothing has to be changed if you don't think it is.

My point is that, technically, the new callback could have run again, say at the end of the build, but the asserts in the test do not check that. They only check a piece of the output, not the whole output.